setuptools/_path.py

```diff
@@ -2,10 +2,10 @@
 import sys
 from typing import Union
 
-if sys.version_info >= (3, 9):
-    StrPath = Union[str, os.PathLike[str]]  #  Same as _typeshed.StrPath
-else:
+if sys.version_info < (3, 9):
     StrPath = Union[str, os.PathLike]
+else:
+    StrPath = Union[str, os.PathLike[str]]  #  Same as _typeshed.StrPath
```

Comment:
I've historically written `> (3, 9)` (or the equivalent `>= (3, 9)`) because I like to put the canonical, preferred code ahead of the legacy code. I don't feel strongly about it and I feel less strongly about it when the compatibility logic is in a `compat` submodule.

My thinking is that if someone is skimming the code, if the legacy code is first, they'll need to read the whole block to get a modern understanding of what's going on, whereas if the preferred code is first, they can stop after reading the first block and just know that the rest is for older Python versions.

This concern illustrates is a case where an `unless` syntax would be so nice to have:

```python
do:
    # preferred behavior
unless sys.version_info < (3, 9):
    # legacy behavior
```

Unfortunately, that's not a thing in Python, so the reader is forced to choose between the more elegant if clause or the more elegant preferred-first ordering.